client/my-sites/customer-home/cards/features/domain-upsell/test/index.jsx

```diff
@@ -0,0 +1,107 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import { render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import nock from 'nock';
+import { Provider } from 'react-redux';
+import configureStore from 'redux-mock-store';
+import DomainUpsell from '../';
+
+jest.mock( 'calypso/state/ui/selectors', () => {
+	return {
+		getSelectedSite: () => {
+			return {
+				ID: 1,
+				plan: 'free',
+			};
+		},
+		getSelectedSiteSlug: () => 'example.wordpress.com',
+	};
+} );
+
+jest.mock( 'calypso/state/sites/domains/selectors', () => {
+	return {
+		getDomainsBySiteId: () => {
+			return [ 'example.wordpress.com' ];
+		},
+	};
+} );
```

Comment:
**Question (non-blocking)**: How hard would it be to set these values in the `initialState` that you pass into the render function below?

Sometimes it's too onerous, and it's just better to mock selectors. But, whenever I can just set redux state directly, I always try to!

This usually gets you a couple of benefits:
1. It expands your test coverage a pinch - it makes sure that the selectors you are using actually select as you expect!
2. In that way, it can actually provide a pinch more flexibility over time. If you refactor the selectors without changing the redux state tree shape, these tests wouldn't need to be updated!

Reply:
Loved your suggestion @dpasque, [did this change here](https://github.com/Automattic/wp-calypso/pull/73391/commits/2e2cc3b03b46a1394f152c0c15ac3e358fe4975e). 

It feels like mocking helps us to write the first iteration of the test, so we don't need to be concerned about the state, but once it's working, changing to state improves a lot the code coverage.

Thanks!